src/map.c

```diff
@@ -3,752 +3,630 @@
  * "LICENSE" for information on usage and redistribution of this file.
  */
 
+/*
+ * This map implementation has undergone extensive modifications, heavily
+ * relying on the rb.h header file from jemalloc. See
+ * https://github.com/jemalloc/jemalloc/blob/dev/include/jemalloc/internal/rb.h
+ * . The original rb.h file served as the foundation and source of inspiration
+ * for adapting and tailoring it specifically for this map implementation.
+ * Therefore, credit and sincere thanks are extended to jemalloc for their
+ * invaluable work.
+ */
+
 #include <assert.h>
-#include <stdint.h>
-#include <stdlib.h>
-#include <string.h>
+#include <memory.h>
 
 #include "map.h"
 
-struct map_internal {
-    struct map_node *head;
-
-    /* Properties */
-    size_t key_size, element_size, size;
+#define RB_MAX_DEPTH (sizeof(void *) << 4)
 
-    map_iter_t it_end, it_most, it_least;
-
-    int (*comparator)(const void *, const void *);
-};
-
-typedef enum { RB_RED = 0, RB_BLACK } map_color_t;
-
-/*
- * Get parent of node
- * @node: pointer to the rb node
- *
- * Return: rb parent node of @node
- */
-static inline map_node_t *rb_parent(const map_node_t *node)
+/* Left accessors */
+static inline map_node_t *rbtn_left_get(map_node_t *node)
```

Comment:
I would like to proposed the following simplifications:
* `rbtn` -> `rbnode` (rbtree node)
* `rbtree` -> `rb` (not necessary to use fully qualified name)

Reply:
In the proposed naming scheme, the above function name would be adjusted as `rbnode_get_left` where `rbnode_` is the prefix followed by certain verbs such as `get` and `set`.